zilliqa/src/exec.rs

```diff
@@ -1323,13 +1328,49 @@ impl From<Account> for PendingAccount {
     }
 }
 
+fn cache_external_libraries(
+    state: &State,
+    ext_libs_cache_dir: &str,
+    ext_libraries: &[ExternalLibrary],
+) -> Result<()> {
+    let ext_libs_path = Path::new(ext_libs_cache_dir);
+
+    for lib in ext_libraries {
+        let account = state.get_account(lib.address)?;
+        match &account.code {
+            Code::Evm(_) => {
+                return Err(anyhow!(
+                    "Impossible to load an EVM contract as a Scilla library."
+                ));
+            }
+            Code::Scilla {
+                code, init_data, ..
+            } => {
+                let contract_init = ContractInit::new(serde_json::from_str(init_data)?)?;
+                if !contract_init.is_library()? {
+                    return Err(anyhow!(
+                        "Impossible to load a non-library contract as a Scilla library."
+                    ));
+                }
+
+                let file_path = ext_libs_path.join(&lib.name);
+                fs::write(&file_path, code).with_context(|| {
```

Comment:
FWIW, this code:

 - will race against anything else running at the time (I think we're safe, though, since there is only one other thing that could be - the scilla interpreter - and that isn't running when this code is executed). Might be worth a comment though.
 -  Doesn't atomically write the file (again, I think we're safe-ish unless disk space runs out or something).
 - Allows a user who can create a library with the appropriate name (`../../../fish/soup.txt`) to overwrite any file on the node's filesystem to which the node's account has access.

I'd think it was worth a comment on why the last one (in particular!) isn't a threat ...